### ISSUE TITLE
how to use "upsert" correctly ?

### ISSUE THREAD

**Issue by ajonno** [supabase-community/supabase-csharp#29]
Am having probs up-serting. 

First here's the equivalent SQL that works fine:
```
INSERT INTO movie (name, release_date, created_at, overview, tmdb_movie_id) VALUES
	('Born on the Fourth of July', '1989-12-19 13:00:00', '2022-07-28 11:29:28.52816+00', 'XXXXXXXX test', 2604)
ON CONFLICT (tmdb_movie_id)
DO UPDATE SET
	name = EXCLUDED.name
	, overview = EXCLUDED.overview;
```

However when using the API I'm getting an error telling me about the "UNIQUE" constraint that is currently set on one of the columns.

The call is simple:
```var movieResponse = await movieHandle!.Upsert(movie);```

but the error message is:
```duplicate key value violates unique constraint "movie_tmdb_movie_id_key"```

From what I can see, it seems like if a table has any field set as "UNIQUE", the upsert api won't work. Unless that is there is maybe a parameter we need to set ? The Dart API has an "onConflict" param which is designed for this purpose but can't see it anywhere for this C# api ?

How do I correctly set things up so that an upsert will succeed ?

for reference this is current model definition:
```
    public class Movie : SupabaseModel
    {
        [PrimaryKey("id", false)]
        public string Id { get; set; }
    
        [Postgrest.Attributes.Column("name")]
        public string Name { get; set; }
    
        [Postgrest.Attributes.Column("overview")]
        public string Overview { get; set; }
    
        [Postgrest.Attributes.Column("release_date")]
        public DateTime? ReleaseDate { get; set; }
    
        [Postgrest.Attributes.Column("tmdb_movie_id")]
        public int TmdbMovieId { get; set; }
        
        [Postgrest.Attributes.Column("created_at")]
        public DateTime CreatedAt { get; set; }
    }
```

**Comment by acupofjose** [comment]
And just to be clear, the `PrimaryKey` Attribute is referencing the Postgrest namespace and not another package?

**Comment by ajonno** [comment]
ohh good spot! hmm let me double back on this and ill confirm

**Comment by ajonno** [comment]
ok checked, yes thats ok in the model. ie the _Postgres.Attributes_.PrimaryKey isn't needed

so im still not sure why the upsert fails. could it be a bug ? (ie relating to the UNIQUE modifier)

**Comment by acupofjose** [comment]
Weird. But okay! Yeah I’ll need to take a look at the `OnConflict` and see if we need to add that to the postgrest api - let me get back to you.

**Comment by acupofjose** [comment]
@ajonno available in `supabase-csharp@5.0.1`! You can set either using either:

```c#
var movieResponse = await movieHandle!.OnConflict("tmdb_movie_id").Upsert(movie);
```

OR 

```c#
var options = new QueryOptions { OnConflict = "tmdb_movie_id" };
var movieResponse = await movieHandle!.Upsert(movie, options);
```
